tools/pepsi/src/progress_indicator.rs

```diff
@@ -0,0 +1,121 @@
+use std::{borrow::Cow, time::Duration};
+
+use color_eyre::{owo_colors::OwoColorize, Report, Result};
+use futures_util::{stream::FuturesUnordered, Future, StreamExt};
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+
+pub struct ProgressIndicator {
+    multi_progress: MultiProgress,
+    default_style: ProgressStyle,
+    error_style: ProgressStyle,
+    success_style: ProgressStyle,
+}
+
+impl ProgressIndicator {
+    pub fn new() -> Self {
+        Self {
+            multi_progress: MultiProgress::new(),
+            default_style: ProgressStyle::with_template("{prefix:.bold.dim} {spinner} {msg}")
+                .unwrap()
+                // The last char is ignored as it provides a final state
+                .tick_chars("⠏⠋⠙⠹⢸⣰⣠⣄⣆⡇ "),
+            error_style: ProgressStyle::with_template("{prefix:.bold.dim} {msg}").unwrap(),
+            success_style: ProgressStyle::with_template("{prefix:.bold.dim} {msg}").unwrap(),
+        }
+    }
+
+    pub fn task(&self, prefix: String) -> Task {
+        let spinner = ProgressBar::new_spinner()
+            .with_style(self.default_style.clone())
+            .with_prefix(format!("[{prefix}]"));
+        spinner.enable_steady_tick(Duration::from_millis(100));
+        Task {
+            progress: self.multi_progress.add(spinner),
+            error_style: self.error_style.clone(),
+            success_style: self.success_style.clone(),
+        }
+    }
+
+    pub async fn map_tasks<T, F, M>(
+        items: impl IntoIterator<Item = T>,
+        message: impl Into<Cow<'static, str>> + Clone,
```

Comment:
Why not?

```suggestion
        message: &'static str,
```

Do we want to expose the type complexity of indicatif or provide a type interface how we are using it in our code base?

Reply:
Thanks for the suggestion, done.